### PR TITLE
New topics UI

### PR DESCRIPTION
**Pull request by jondewijones**
### What

Add new topics selector - [DIS-3824](https://officefornationalstatistics.atlassian.net/jira/software/c/projects/DIS/boards/1822?selectedIssue=DIS-3824)
Add new canonical "main" topic selector - [DIS-4525](https://officefornationalstatistics.atlassian.net/jira/software/c/projects/DIS/boards/1822?selectedIssue=DIS-4525)

⚠️ Note:
- Standard radio and checkbox (rather than design-system react component) components have been used to avoid bugs found with these components in previous topics work where direct DOM manipulation was required to make certain things work
- The arrow/chevron icon for the accordion component doesn't work/show. Unsure why but felt out of scope for this ticket to invest lots of time to fix. If this is a priority then a bug ticket can be raised to look at separately 

### How to review

Try create and edit a series and add/edit topics accordingly. It's worth creating a fresh series to do this with since seeded stack data (e.g. `make up-with-seed`) uses top level topics, whereas the new topics UI is focused on sub topics. 

Once successfully created or edited topics data should appear in dataset-api as before e.g. `"topics": ["id-001", "id-002"]`

### Who can review

Anyone

### CODE REVIEW

File: src/components/topics/topicsData.js
Comment (JoshSinghONS) [inline src/components/topics/topicsData.js:1]: I'd be tempted to remove this file and move these functions into src/utils/topics/topics.js

File: src/components/accordion/AccordionItem.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    const handleOnClick = () => {
+        setIsOpen(!isOpen);
+    };
+
+    const idPrefix = id ? `${id}-` : "";
+    const sanitisedId = sanitiseString(`${idPrefix}accordion-item-${accordionItem.id}`);
+    const sanitisedDataTestId = sanitiseString(`${idPrefix}accordion-item-${accordionItem.id}`);
+
+    return (
+        <div id={sanitisedId} className="ons-details ons-details--accordion" data-group="accordion-example" key={accordionItem.id} data-testid={sanitisedDataTestId}>
```
Comment (JoshSinghONS) [inline src/components/accordion/AccordionItem.js]: What's a data-group? Also I note it's got example in the value for it. Left over typo?

File: src/components/accordion/AccordionItem.test.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import "@testing-library/jest-dom";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+
+import AccordionItem from "./AccordionItem";
+
+describe("AccordionItem", () => {
+    test("renders closed by default and toggles content open and closed", async () => {
+        const user = userEvent.setup();
```
Comment (JoshSinghONS) [inline src/components/accordion/AccordionItem.test.js]: This is new, what's this do?

File: tests/component/seriesCreate.spec.js
Comment (JoshSinghONS) [inline tests/component/seriesCreate.spec.js:1]: Should probably add a test for the Show errors on mandatory fields section for topics

File: src/components/topics/Topics.js [hunk 3]
```diff
@@ -1,91 +1,202 @@
...
+            if (sub) return sub.label;
+        }
+    }
+    return null;
+};
 
 export default function Topics({ listOfAllTopics, preSelectedTopics, topicsError }) {
-    const [selectedTopics, setSelectedTopics] = useState(preSelectedTopics);
-    const [checkboxOptionsItems, setCheckboxOptionsItems] = useState(createCheckboxes);
+    const [selectedTopics, setSelectedTopics] = useState(() => preSelectedTopics || []);
```
Comment (JoshSinghONS) [inline src/components/topics/Topics.js:38]: So SeriesForm has 

`export default function SeriesForm({ currentTitle = "", currentID = "", currentDescription = "", currentTopics = [], currentQMI = "", currentKeywords = "", currentContacts = [], listOfAllTopics, action }) {`

and

`<Topics
                    listOfAllTopics={listOfAllTopics}
                    preSelectedTopics={formState.submission?.originalTopics ?? currentTopics}
                    topicsError={(formState.errors && formState.errors.topics) ? formState.errors.topics : null}
                />`

So not sure if it's required for selectedTopics to need the `|| []` because SeriesForm looks to sort that out for you?

File: src/components/topics/topics.test.js
Comment (JoshSinghONS) [inline src/components/topics/topics.test.js:1]: I'd probably add another test for removing a topic from a list of selected topics.

File: src/components/topics/topicsData.js
Comment (jondewijones) [reply src/components/topics/topicsData.js:1]: I had this debate with myself over and over. It feels like the functions in this file are quite specific to the component as they map the data to how we need it for the component rather than a generic get-some-data type function whereas the functions in `utils/*` are more generic and hence are in utils. However, it feels a little weird having them separated and couldn't come to a decisive decision so in the end I stopped debating with myself and left them were they were.  

Ultimately I feel like they should reside in the same place but neither of the current places feel right for both types 😂 . If you have particularly strong feelings I can move it.

File: src/components/accordion/AccordionItem.js (hunk 1 above)
Comment (jondewijones) [reply src/components/accordion/AccordionItem.js]: Ahh this was left over from copying the HTML for the accordion component from the [design system ](https://service-manual.ons.gov.uk/design-system/components/accordion). I've removed it as it's not needed. I think it's only there for some JS functionality which we don't need as React is handling that for us.

File: src/components/accordion/AccordionItem.test.js (hunk 2 above)
Comment (jondewijones) [reply src/components/accordion/AccordionItem.test.js]: You mean the `userEvent.setup()`? It's the "recommended" way of simulating browser behaviours like clicks. It's effectively the same as `userEvent.click()` but instead we'd `user.click()` after setting up the `user` instance. 

You can read more about it on this [userEvent discussion post](https://github.com/testing-library/user-event/discussions/1036).

It seems to only be useful in a handful of edge cases so I've removed it and kept things consistent with other test files.

File: tests/component/seriesCreate.spec.js
Comment (jondewijones) [reply tests/component/seriesCreate.spec.js:1]: I wrongly assumed these already existed. I have added them in.

File: src/components/topics/Topics.js (hunk 3 above)
Comment (jondewijones) [reply src/components/topics/Topics.js:38]: `Topics` has its own `preSelectedTopics || []` logic as defensive mechanism, see https://jumpcloud.com/it-index/what-is-defensive-coding.

As Topics is a component it could be called from anywhere. Theoretically another page could call the `<Topics />` component and not contain the logic that SeriesForm has. So what appears as duplicate logic is more of a failsafe, makes the component more independent (i.e. doesn't rely on the call to perform some pre-logic) and therefore more robust. 

If we were to remove any duplication we'd probably want to remove it from SeriesForm.

File: src/components/topics/topics.test.js
Comment (jondewijones) [reply src/components/topics/topics.test.js:1]: Good spot. Added a test for this

**Review by JoshSinghONS — APPROVED** [review]
